src/devices/machine/mc68328.h

```diff
@@ -401,513 +378,425 @@ class mc68328_device : public m68000_device
+	virtual void register_state_save();
+
+	void update_ipr_state(u32 changed_mask);
+	void update_imr_state(u32 changed_mask);
+	void set_interrupt_line(u32 line, u32 active);
+	virtual u32 get_irq_mask_for_level(int level) = 0;
+	virtual int get_irq_level_for_mask(u32 level) = 0;
+
+	virtual emu_timer *get_timer(int timer) = 0;
+	virtual timer_regs &get_timer_regs(int timer) = 0;
+	virtual u32 get_timer_int(int timer) = 0;
+	template<int Timer> u32 get_timer_frequency();
+	template<int Timer> void update_gptimer_state();
+	template<int Timer> TIMER_CALLBACK_MEMBER(timer_tick);
 
-	void register_state_save();
+	virtual u16 lcd_get_lxmax_mask() = 0;
+	virtual u32 lcd_get_line_word_count() = 0;
+	virtual attotime lcd_get_line_rate() = 0;
+	virtual u8 lcd_get_panel_bit_size() = 0;
+	virtual attotime get_pixclk_rate() = 0;
 
-	template<int Timer> TIMER_CALLBACK_MEMBER(timer_tick);
-	TIMER_CALLBACK_MEMBER(pwm_tick);
+	TIMER_CALLBACK_MEMBER(refclk_tick);
+	virtual TIMER_CALLBACK_MEMBER(pwm_tick) = 0;
 	TIMER_CALLBACK_MEMBER(rtc_tick);
 	TIMER_CALLBACK_MEMBER(spim_tick);
 	TIMER_CALLBACK_MEMBER(lcd_scan_tick);
 	void fill_lcd_dma_buffer();
 
-	emu_timer *m_gptimer[2];
-	emu_timer *m_rtc;
+	virtual bool rtc_int_is_active() = 0;
+	virtual void rtc_advance_seconds();
+	virtual u16 rtc_get_int_mask() = 0;
+	virtual bool rtc_get_alarm_match() = 0;
+
+	emu_timer *m_refclk;
 	emu_timer *m_pwm;
+	emu_timer *m_rtc;
 	emu_timer *m_spim;
 
 	emu_timer *m_lcd_scan;
 	bool m_lcd_first_line;
-	uint32_t m_lcd_sysmem_ptr;
-	std::unique_ptr<uint16_t[]> m_lcd_line_buffer;
-	uint32_t m_lcd_line_bit;
-	uint32_t m_lcd_line_word;
+	u32  m_lcd_sysmem_ptr;
+	u32  m_lcd_line_bit;
+	u32  m_lcd_line_word;
 	bool m_lsclk;
-
-	devcb_write8  m_out_port_a_cb;
-	devcb_write8  m_out_port_b_cb;
-	devcb_write8  m_out_port_c_cb;
-	devcb_write8  m_out_port_d_cb;
-	devcb_write8  m_out_port_e_cb;
-	devcb_write8  m_out_port_f_cb;
-	devcb_write8  m_out_port_g_cb;
-	devcb_write8  m_out_port_j_cb;
-	devcb_write8  m_out_port_k_cb;
-	devcb_write8  m_out_port_m_cb;
-
-	devcb_read8   m_in_port_a_cb;
-	devcb_read8   m_in_port_b_cb;
-	devcb_read8   m_in_port_c_cb;
-	devcb_read8   m_in_port_d_cb;
-	devcb_read8   m_in_port_e_cb;
-	devcb_read8   m_in_port_f_cb;
-	devcb_read8   m_in_port_g_cb;
-	devcb_read8   m_in_port_j_cb;
-	devcb_read8   m_in_port_k_cb;
-	devcb_read8   m_in_port_m_cb;
+	std::unique_ptr<u16[]> m_lcd_line_buffer;
+
+	devcb_write_line    m_out_port_a_cb[8];
+	devcb_write_line    m_out_port_b_cb[8];
+	devcb_write_line    m_out_port_c_cb[8];
+	devcb_write_line    m_out_port_d_cb[8];
+	devcb_write_line    m_out_port_e_cb[8];
+	devcb_write_line    m_out_port_f_cb[8];
+	devcb_write_line    m_out_port_g_cb[8];
+
+	devcb_read_line     m_in_port_a_cb[8];
+	devcb_read_line     m_in_port_b_cb[8];
+	devcb_read_line     m_in_port_c_cb[8];
+	devcb_read_line     m_in_port_d_cb[8];
+	devcb_read_line     m_in_port_e_cb[8];
+	devcb_read_line     m_in_port_f_cb[8];
+	devcb_read_line     m_in_port_g_cb[8];
```

Comment:
Could you use `devcb_write_line::array<8>` and `devcb_read_line::array<8>` for these?  It makes the constructor cleaner because you only need a single `*this`, and you can use `resolve_all` or `resolve_all_safe` rather than resolving them one at a time in a loop.  Example in src/devices/machine/output_latch.{h,cpp} or the SA1110 and SA1111 devices.